Lib/functools.py

```diff
@@ -20,6 +20,7 @@
 from reprlib import recursive_repr
 from _thread import RLock
 from types import GenericAlias
+from warnings import warn
```

Comment:
Should import this inside the `if lock:` block in `cached_property.__init__`, so that users only pay the performance cost for importing the module if they're using the deprecated behaviour